src/Features/Core/Portable/UnifiedSuggestions/UnifiedSuggestedActions/UnifiedCodeFixSuggestedAction.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+
+namespace Microsoft.CodeAnalysis.UnifiedSuggestions
+{
+    /// <summary>
+    /// Similar to UnifiedCodeFixSuggestionAction, but in a location that can be used by
```

Comment:
> UnifiedCodeFixSuggestionAction

Typo? Did you mean `CodeFixSuggestionAction`?

Reply:
`CodeFixSuggestionAction` is an internal type. Instead of creating a duplicate, would it be better to move it to a shared project such that when included in LSP type it's base type `SuggestedAction` actually uses `UnifiedSuggestedAction`?

For example, consider the below type, which if required to be moved to shared project can be re-written as follows and avoid creating more cloned code: https://github.com/dotnet/roslyn/blob/e704ca635bd6de70a0250e34c4567c7a28fa9f6d/src/EditorFeatures/Core.Wpf/Suggestions/PreviewChanges/PreviewChangesSuggestedAction.cs#L12-L19

```csharp
#if LSP
using SuggestedAction = UnifiedSuggestedAction;
#endif

namespace Microsoft.CodeAnalysis.Editor.Implementation.Suggestions
{
    internal partial class SuggestedActionWithNestedFlavors
    {
        /// <summary>
        /// Suggested action for showing the preview-changes dialog.  Note: this is only used
        /// as a 'flavor' inside CodeFixSuggestionAction and CodeRefactoringSuggestedAction.
        /// </summary>
        private sealed partial class PreviewChangesSuggestedAction : SuggestedAction
        {
```

Reply:
Ah, seems like `CodeFixSuggestedAction` actually uses bunch of EditorFeatures layer APIs, so my prior suggestion will not work. It would still be good to have both `CodeFixSuggestedAction` and `UnifiedCodeFixSuggestionAction` implement a common internal interface `ICodeFixSuggestedAction` to ensure they always stay in sync.